packages/graphback-runtime/src/data/KnexDBDataProvider.ts

```diff
@@ -20,29 +20,37 @@ export class KnexDBDataProvider<Type = any, GraphbackContext = any> implements G
   protected db: Knex;
   protected baseType: GraphQLObjectType;
   protected tableName: string;
+  protected tableMap: ModelTableMap;
 
   constructor(baseType: GraphQLObjectType, db: Knex) {
     this.db = db;
     this.baseType = baseType;
     // TODO build and use mapping here
-    this.tableName = lowerCaseFirstChar(baseType.name);
+    this.tableMap = buildModelTableMap(baseType);
+    this.tableName = this.tableMap.tableName;
   }
 
   public async create(data: Type): Promise<Type> {
-    // TODO refactor to use new knex api
-    const [id] = await this.db(this.tableName).insert(data);
-    const dbResult = await this.db.select().from(this.tableName).where('id', '=', id)
+    const { data: createData } = getDatabaseArguments(this.tableMap, data);
+
+    // tslint:disable-next-line: await-promise
+    const [id] = await this.db(this.tableName).insert(createData);
```

Comment:
I do not think that this will work with anything else than `id`. 
We need to specify returning as in docs: http://knexjs.org/#Builder-insert

Reply:
Ok. Ignore me. That will not work with MySQL. We need to update CRUD tests to use non id and mapping to tests this use case